packages/core/src/Layout/Layout.js

```diff
@@ -59,26 +52,23 @@ class Layout {
       // eslint-disable-next-line no-await-in-loop
       this.layoutProcessed = await nodeProcessor.process(this.sourceFilePath, nunjucksProcessed,
                                                          this.sourceFilePath, layoutVariables);
+      this.layoutPageNavUuid = nodeProcessor.pageNavProcessor.getUuid();
+
       this.layoutProcessed = pluginManager.postRender(nodeProcessor.frontMatter, this.layoutProcessed);
 
       const pageBodyVarRegex = new RegExp(this.layoutPageBodyVariable, 'g');
-      const pageNavVarRegex = new RegExp(this.layoutPageNavVariable, 'g');
       const bodyVarMatch = this.layoutProcessed.match(pageBodyVarRegex);
-      const pageNavVarMatch = this.layoutProcessed.match(pageNavVarRegex);
       numBodyVars = bodyVarMatch ? bodyVarMatch.length : 0;
-      numPageNavVars = pageNavVarMatch ? pageNavVarMatch.length : 0;
 
       triesLeft -= 1;
-    } while (triesLeft > 0 && (numBodyVars > 1 || numPageNavVars > 1));
+    } while (triesLeft > 0 && numBodyVars > 1);
```

Comment:
Just to confirm my understanding of this "hacky" implementation, with the new changes, we won't actually need this "retries" loop anymore, right? 

Considering that we only have 1 variable left -- `LAYOUT_PAGE_BODY_VARIABLE`, so there won't ever be false-positives where `uuidv4` generates the same uuid for 2 different variables (from my inference, the 10 retries is to handle this scenario of false positives). 

This is also to say that any `bodyVarMatch` with count >= 2 will always be duplicates. 

However, we will leave the existing implementation as it is for now, in case there will be new variables needed in the future.

Reply:
@wxwxwxwx9 Its still possible if the user content contains the generated uuid, but really shouldn't happen in a lifetime.
The loop guards against that nevertheless.

Its a rather hacky approach, but this flow allows processing pages and layouts independently for performance. (also, architecture / code quality wise)

Reply:
> Its a rather hacky approach

something we should fix, but haven't been able to think of a good alternative for a while 🤔

Reply:
> > Its a rather hacky approach
> 
> something we should fix, but haven't been able to think of a good alternative for a while 🤔

Hmm yeah this would require some thinking.